src/logic/currentSession/store/reducer/currentSession.ts

```diff
@@ -5,17 +5,20 @@ import { UPDATE_VIEWED_SAFES } from 'src/logic/currentSession/store/actions/upda
 import { CLEAR_CURRENT_SESSION } from 'src/logic/currentSession/store/actions/clearCurrentSession'
 import { saveCurrentSessionToStorage } from 'src/logic/currentSession/utils'
 import { REMOVE_VIEWED_SAFE } from '../actions/removeViewedSafe'
+import { ADD_CURRENT_SAFE_ADDRESS } from 'src/logic/currentSession/store/actions/addCurrentSafeAddress'
 
 export const CURRENT_SESSION_REDUCER_ID = 'currentSession'
 const MAX_VIEWED_SAFES = 10
 
 export type CurrentSessionState = {
   viewedSafes: string[]
+  currentSafeAddress: string
```

Comment:
Agreed, I think its worthwile to save the whole `chains` response in the store instead of caching it in a global namespace. We could then replace `getShortName` and similar functions with selectors. I would suggest we do it in a separate task though.